src/node-lib/curriculum-api-2023/queries/subjectListing/constructSubjectsFromLessonData.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  subjects,
+  subjectSlugs,
+  SyntheticUnitvariantLessons,
+} from "@oaknational/oak-curriculum-schema";
+
+interface UnprocessedSubject {
+  [key: string]: {
+    subjectTitle: typeof subjects | string;
+    subjectSlug: typeof subjectSlugs | string;
+    programmeSlug: string;
+    unitIds: Set<number>;
+    lessonSlugs: Set<string>;
+    programmeSlugs: Set<string>;
+  };
+}
+
+interface ProcessedSubject {
+  subjectTitle: typeof subjects | string;
+  subjectSlug: typeof subjectSlugs | string;
+  programmeSlug: string;
+  unitCount: number;
+  lessonCount: number;
+  programmeCount: number;
+}
+
+export const constructSubjectsFromLessonData = (
+  lessons: SyntheticUnitvariantLessons[],
+): ProcessedSubject[] => {
+  const subjects = lessons.reduce((acc, lesson) => {
+    // eslint-disable-next-line prefer-const
+    let { programme_slug, lesson_slug } = lesson;
+    const {
+      subject,
+      keystage_slug,
+      phase,
+      subject_slug,
+      tier_slug,
+      examboard_slug,
+    } = lesson.programme_fields;
+    const { unit_id } = lesson.unit_data;
+    const originalProgrammeSlug = programme_slug;
+
+    //remove programme factors from new data
+    if (tier_slug || examboard_slug) {
+      const baseProgrammeSlug = `${subject_slug}-${phase}-${keystage_slug}`;
```

Comment:
don't think this needs to be conditional does it? we can create `baseProgrammeSlug` up top and use that as the key in all cases instead of `programme_slug`